src/controllers/inspectionpolicy_controller.go

```diff
@@ -128,12 +128,14 @@ func (r *InspectionPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	policy.Spec.WorkNamespace = wns
 
 	// Ensure Setting is correctly configured.
-	datasource, err := r.EnsureSettings(ctx, policy)
-	if err != nil {
-		log.Error(err, "unable to ensure the settings in inspection policy")
-		return ctrl.Result{}, err
+	if policy.Spec.Inspector.WorkloadScannerImage == "" {
```

Comment:
The if condition should be that any of  other images are != "" (just in case there are more than one scanners defined in the inspection policy)

Reply:
Yes, to be precise.  SettingsName is used only in Image Scanner.
```
if policy.Spec.Inspector.Image != "" {
          datasource, err := r.EnsureSettings(ctx, policy)
         if err != nil {
		log.Error(err, "unable to ensure the settings in inspection policy")
		return ctrl.Result{}, err
	}
}
```

Reply:
Currently, they don't have any check for this particular **policy.Spec.Inspector.Image**. Instead, they are ensuring for all other images as well like kubebench and riskImage. 
So, we do not need to change their current logic. Only we can escape for workloadscanner.

Reply:
But that's not right because if there are more than one image (for example, workloadscanner and inspector) we are skipping it, while it would be required for the inspector scanner.

That's why I suggested to check if any of the other images was != "". @bgubbala suggested we could optimize that by only checking when the Inspector scanner is used. I guess this is perfectly valid as long as Harbor is only being used by the Inspector Scanner (which I think it's true).

Reply:
@snootan  There was one more` EnsureSettings` under `src/pkg/inspectors/imagescanner/pkg/controller.go`
It is being used only in image scanner.

```
func (c *controller) EnsureSettings(ctx context.Context, policy *v1alpha1.InspectionPolicy) (*v1alpha1.Setting, error) {
	settingsName := policy.Spec.SettingsName
	if settingsName == "" {
		return nil, errors.New("Invalid settings name")
	}
	setting := &v1alpha1.Setting{}
	namespaced := types.NamespacedName{
		Name: settingsName,
	}
```

Reply:
Check with a config with all the scanners running and verify there were no restarts across the manager and scanners.

Reply:
As we are using this setting in risk manager as well. So, we are not going to change any thing in current logic. 
Same discussed with Pablo as well.
@bgubbala, above setting is getting used for imagescanner. So, it would be OK.